### PR TITLE
Switch to Rust stable

### PR DESCRIPTION
**Pull request by fridrik01**
Since we have now removed substrate-wasm-builder dependency we are able to switch to rust stable channel.

I needed to update some of the integration tests that dependended on unstable nightly features.

Fixes: https://github.com/filecoin-project/ref-fvm/issues/1717

### CODE REVIEW

**Review by vyzo — APPROVED** [review]
at last

File: testing/integration/tests/gas_calibration_test.rs [hunk 1]
```diff
@@ -142,10 +140,7 @@ fn on_event_evm_shapes() {
     }
 
     for (obs, name) in vec![(validate_obs, CHARGE_VALIDATE), (accept_obs, CHARGE_ACCEPT)].iter() {
-        let regression = obs
-            .group_by(|a, b| a.label == b.label)
-            .map(|g| least_squares(g[0].label.to_owned(), g, 0))
-            .collect::<Vec<_>>();
+        let regression = run_linear_regression(&obs);
```
Comment (maciejwitowski) [inline testing/integration/tests/gas_calibration_test.rs:143]: Just curious, is this an opportunistic refactor or is it somehow related to switching to Rust stable?

Comment (fridrik01) [reply testing/integration/tests/gas_calibration_test.rs:143]: Its a required rewrite to get rid of the `#![feature(slice_group_by)]` which requires rust nightly to compile

**Comment by fridrik01** [comment]
Updated the PR to be merged into master, requires re-review

**Comment by fridrik01** [comment]
I rebased this against master and ran cargo clippy locally since its been a week since this PR was updated.

Now I get (I didn't see this before) that the storage-proofs-porep crate can not be compiled since its using a feature not available in the stable channel (interestingly this doesn't show up in CI).

See:

```
> cargo clippy --locked --all --all-targets --no-default-features
info: syncing channel updates for 'stable-aarch64-apple-darwin'
info: latest update on 2023-03-23, rust version 1.68.1 (8460ca823 2023-03-20)
info: downloading component 'clippy'
info: downloading component 'llvm-tools'
...
   Compiling serde_derive v1.0.152
    Checking cfg-if v1.0.0
   Compiling memchr v2.5.0
...
   Compiling psm v0.1.21
    Checking wasmtime-asm-macros v2.0.2
error[E0554]: `#![feature]` may not be used on the stable release channel
 --> /Users/fridrik/.cargo/registry/src/github.com-1ecc6299db9ec823/storage-proofs-porep-13.0.0/src/lib.rs:3:46
  |
3 | #![cfg_attr(target_arch = "aarch64", feature(stdsimd))]
  |                                              ^^^^^^^

   Compiling async-lock v2.6.0
    Checking wasmparser v0.99.0
    Checking fvm_ipld_hamt v0.6.1 (/Users/fridrik/workspace/ref-fvm/ipld/hamt)
    Checking fvm_ipld_amt v0.5.1 (/Users/fridrik/workspace/ref-fvm/ipld/amt)
    Checking addr2line v0.17.0
For more information about this error, try `rustc --explain E0554`.
error: could not compile `storage-proofs-porep` due to previous error
```

**Review by Stebalien — APPROVED** [review]
🥳

**Comment by Stebalien** [comment]
Well, except that clippy just really hates everyone.

**Pull request merged**